### ISSUE TITLE
--namespace cli argument needed for BMO

### ISSUE THREAD

**Issue by maelk** [metal3-io/baremetal-operator#464]
In order to fulfill the provider contract, BMO must accept a --namespace cli argument instead of using a WATCH_NAMESPACE environment variable : https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html

We would propose to not fail if the environment variable WATCH_NAMESPACE is unset, and override it with --namespace if provided in the CLI.

**Comment by fmuyassarov** [comment]
isn't --namespace option already there?

```
Flags:
      --enable-delve            Start the operator using the delve debugger
      --go-ldflags string       Set Go linker options
  -h, --help                    help for local
      --kubeconfig string       The file path to kubernetes configuration file; defaults to location specified by $KUBECONFIG with a fallback to $HOME/.kube/config if not set
      --namespace string        The namespace where the operator watches for changes.
      --operator-flags string   The flags that the operator needs. Example: "--flag1 value1 --flag2=value2"
```

**Comment by maelk** [comment]
Those are the operator-sdk options, not the manager options

**Comment by dhellmann** [comment]
The namespace to use is determined in the main program for the controller at https://github.com/metal3-io/baremetal-operator/blob/master/cmd/manager/main.go#L45

**Comment by fmuyassarov** [comment]
Fix in  #471
